ansible/roles/test/files/ptftests/wr_arp.py

```diff
@@ -242,21 +253,22 @@ def runTest(self):
             self.assertTrue(False, "DUT returned error for first uptime request")
 
         records = defaultdict(dict)
-        stop_at = time.time() + self.how_long
-        rebooted = False
-        while time.time() < stop_at:
-            for test in self.tests:
-                for port in test['acc_ports']:
-                    nr_rcvd = self.testPort(port)
-                    records[port][time.time()] = nr_rcvd
-            if not rebooted:
-                result = self.req_dut('WR')
-                if result.startswith('ok'):
-                    rebooted = True
-                else:
-                    self.log("Error in WR")
-                    self.req_dut('quit')
-                    self.assertTrue(False, "Error in WR")
+        self.stop_at = time.time() + self.how_long
+
+        test_port_thr = threading.Thread(target=self.test_port_thr)
+        test_port_thr.setDaemon(True)
+        test_port_thr.start()
+
+        self.log("Issuing WR command")
+        result = self.req_dut('WR')
+        if result.startswith('ok'):
```

Comment:
is there a change to warm-reboot command that would make it blocking. I believe the idea is to have warm-reboot run while having ARP packet flowing. If warm-reboot is taking long to kick in, this could defeat the test case scenario by offloading ARP packet to a thread of its own. @pavel-shirshov is this change OK?

Reply:
Yes, but in fact test waits about 50 secs for warm-reboot result receiving and during this time the ARPs are not sent which makes the test failing even on the old scripts.

Reply:
> I believe the idea is to have warm-reboot run while having ARP packet flowing.

The changes perform exactly this scenario.

Reply:
I agree. My concern is that the test is sensitive to the test requested run time (`how_log`.) If it is too short, we may not hit the scenario we are testing for. I understand it is not related to this change at hand here. Can we assert that the test `self.stop_at` is greater than the current time right after the warm-reboot command finishes. This way we could make sure ARPing was running during the warm-reboot process.

Reply:
Done